src/nnsDapp.ts

```diff
@@ -0,0 +1,98 @@
+import { Actor } from "@dfinity/agent";
+import { idlFactory as certifiedIdlFactory } from "../candid/nns_dapp.certified.idl";
+import { idlFactory, NNSDappService } from "../candid/nns_dapp.idl";
+import {
+  CreateSubAccountResponse,
+  SubAccountDetails,
+} from "../candid/nns_dappTypes";
+import { AccountIdentifier } from "./account_identifier";
+import { AccountNotFoundError } from "./errors";
+import { NNSDappCanisterOptions } from "./types/nnsDapp";
+import { defaultAgent } from "./utils/agent.utils";
+
+export class NNSDappCanister {
+  private constructor(
+    private readonly service: NNSDappService,
+    private readonly certifiedService: NNSDappService
+  ) {
+    this.service = service;
+    this.certifiedService = certifiedService;
+  }
+
+  public static create(options: NNSDappCanisterOptions) {
+    const agent = options.agent ?? defaultAgent();
+    const canisterId = options.canisterId;
+
+    const service =
+      options.serviceOverride ??
+      Actor.createActor<NNSDappService>(idlFactory, {
+        agent,
+        canisterId,
+      });
+
+    const certifiedService =
+      options.certifiedServiceOverride ??
+      Actor.createActor<NNSDappService>(certifiedIdlFactory, {
+        agent,
+        canisterId,
+      });
+
+    return new NNSDappCanister(service, certifiedService);
+  }
+
+  /**
+   * Used to be able to create subAccounts
+   *
+   * TODO: Understand why we need this?
```

Comment:
can we solve these TODO before merging?

nns-js is a public lib, if we add and then remove a function in two builds, that's a breaking change.

Reply:
Good point, but I am concerned about slowing down.  It might make sense to have a convention for naming unstable functions.  E.g. any function with a name starting with _ is considered unstable and its removal or alteration will not be considered a breaking change.  This would give us space to experiment while still having solid guarantees for the stable API.